<p>
    This rule checks that the service layers of different components don't depend on each other.
    Allowed dependencies from the service layer:
    <ul type='disc'>
        <li>Logic layer of the same component</li>
        <li>Common layer</li>
    </ul>
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.<span style='color: #785EF0'>compA.service</span>.api;

    import com.devonfw.ide.sonarqube.<span style='color: #FFB000'><del>compB.service</del></span>.api.Bar;

    public class MyClass {}
</pre>