<p>
    This rule checks that the logic layer of one component does not depend on the dataaccess layer of another component.
    Allowed dependencies from the logic layer: 
    <ul type='disc'>
        <li>Logic layer of a different component</li>
        <li>Dataaccess layer of the same component</li>
        <li>Common layer</li>
    </ul>
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.<span style='color: #785EF0'>compA.logic</span>.api;

    import com.devonfw.ide.sonarqube.<del style='color: #979696'><span style='color: #FFB000'>compB.dataaccess</span></del>.api.Bar;

    public class MyClass {}
</pre>