<p>
    This rule checks that the @Transactional annotation is properly used from JEE.
    The use of the org.springframework.transaction.annotation.Transactional annotation is discouraged,
    use the JEE standard javax.transaction.Transactional. Do not use this annotation in the api scope but to annotate implementations.
</p>
<h2>Noncompliant Code Examples</h2>
<pre>
    package com.devonfw.ide.sonarqube.logic.impl;

    <span style='color: #FFB000'><del>import org.springframework.transaction.annotation.Transactional;</del></span>

    public class MyClass {}
</pre>
<br>
<pre>
    package com.devonfw.ide.sonarqube.common.<span style='color: #FFB000'><del>api</del></span>;

    <span style='color: #FFB000'><del>import javax.transaction.Transactional;</del></span>

    public class MyClass {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    package com.devonfw.ide.sonarqube.logic.<span style='color: #648FFF'><u>impl</u></span>;

    <span style='color: #648FFF'><u>import javax.transaction.Transactional;</u></span>

    public class MyClass {}
</pre>