<p>
    This rule checks that the @Transactional annotation is properly used from JEE.
    The use of the org.springframework.transaction.annotation.Transactional annotation is discouraged,
    use the JEE standard javax.transaction.Transactional. Do not use this annotation in the api scope but to annotate implementations.
</p>
<h2>Noncompliant Code Examples</h2>
<pre>
    package com.devonfw.ide.sonarqube.logic.impl;

    <del style='color: #979696'><span style='color: #FFB000'>import org.springframework.transaction.annotation.Transactional;</span></del>

    public class MyClass {}
</pre>
<br>
<pre>
    package com.devonfw.ide.sonarqube.common.<del style='color: #979696'><span style='color: #FFB000'>api</span></del>;

    <del style='color: #979696'><span style='color: #FFB000'>import javax.transaction.Transactional;</span></del>

    public class MyClass {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    package com.devonfw.ide.sonarqube.logic.<u style='color: #979696'><span style='color: #648FFF'>impl</span></u>;

    <u style='color: #979696'><span style='color: #648FFF'>import javax.transaction.Transactional;</span></u>

    public class MyClass {}
</pre>