<p>This rule checks that QueryDSL is used properly rather than the legacy mysema API.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.dataaccess.api;

    <del stlye='color: #979696'><span style='color: #FFB000'>import com.mysema.query.jpa.impl.JPAQuery;</span></del>
    
    public class MyClass {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    package com.devonfw.ide.sonarqube.dataaccess.api;

    <u style='color: #979696'><span style='color: #648FFF'>import com.querydsl.jpa.impl.JPAQuery;</span></u>

    public class MyClass {}
</pre>