<p>This rule checks that the JPA is only used in the dataaccess layer and optionally in common for embeddables.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.<span style='color: #FFB000'><del>service</del></span>.api;

    <span style='color: #FFB000'><del>import javax.persistence.EntityManager;</del></span>

    public class MyClass {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    package com.devonfw.ide.sonarqube.<u style='color: #979696'></u><span style='color: #648FFF'>dataaccess</u></span>.api;

    <span style='color: #648FFF'><u>import javax.persistence.Entity;</u></span>
    <span style='color: #648FFF'><u>import javax.persistence.Table;</u></span>

    @Entity
    @Table(name="Foo")
    public class FooEntity {}
</pre>