<p>This rule checks that the JPA is only used in the dataaccess layer and optionally in common for embeddables.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.<del style='color: #979696'><span style='color: #FFB000'>service</span></del>.api;

    <del style='color: #979696'><span style='color: #FFB000'>import javax.persistence.EntityManager;</span></del>

    public class MyClass {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    package com.devonfw.ide.sonarqube.<u style='color: #979696'></u><span style='color: #648FFF'>dataaccess</span></u>.api;

    <u style='color: #979696'><span style='color: #648FFF'>import javax.persistence.Entity;</span></u>
    <u style='color: #979696'><span style='color: #648FFF'>import javax.persistence.Table;</span></u>

    @Entity
    @Table(name="Foo")
    public class FooEntity {}
</pre>