<p>
    This rule checks that Hibernate is only used in the dataaccess layer. In any other layer, JPA should be preferred.
    Only the use of the org.hibernate.validator package is allowed outside of the dataaccess layer.
    Regardless of in which layer they are used, the following Hibernate annotations are forbidden:
    <ul type='disc'>
        <li>org.hibernate.annotations.OrderBy</li>
        <li>org.hibernate.annotations.Entity</li>
        <li>org.hibernate.annotations.AccessType</li>
        <li>org.hibernate.annotations.ForeignKey</li>
        <li>org.hibernate.annotations.Cascade</li>
        <li>org.hibernate.annotations.Index</li>
        <li>org.hibernate.annotations.IndexColumn</li>
    </ul>
    Also, Hibernate enver implementations and Hibernate internals should never be used directly or outside of the impl scope of a dataaccess layer.
</p>
<h2>Noncompliant Code Examples</h2>
<pre>
    package com.devonfw.ide.sonarqube.<del style='color: #979696'><span style='color: #FFB000'>logic</span></del>.impl;

    <del style='color: #979696'><span style='color: #FFB000'>import org.hibernate.Session;</span></del>

    public class MyClass {}
</pre>
<br>
<pre>
    package com.devonfw.ide.sonarqube.dataaccess.api;

    <del style='color: #979696'><span style='color: #FFB000'>import org.hibernate.annotations.OrderBy;</span></del>

    public class MyClass {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    package com.devonfw.ide.sonarqube.<u style='color: #979696'><span style='color: #648FFF'>dataaccess.impl</span></u>;

    <u style='color: #979696'><span style='color: #648FFF'>import org.hibernate.Session;</span></u>
    <u style='color: #979696'><span style='color: #648FFF'>import org.hibernate.annotations.FilterDef;</span></u>
    <u style='color: #979696'><span style='color: #648FFF'>import org.hibernate.envers.Audited;</span></u>

    public class MyClass {}
</pre>