<p>This rule checks that Objects is used from JDK.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.dataaccess.api;

    <span style='color: #FFB000'><del>import com.google.common.base.Objects;</del></span>

    public class MyClass {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    package com.devonfw.ide.sonarqube.dataaccess.api;

    <span style='color: #648FFF'><u>import java.util.Objects;</u></span>

    public class MyClass {}
</pre>