<p>This rule checks that Objects is used from JDK.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.dataaccess.api;

    <del style='color: #979696'><span style='color: #FFB000'>import com.google.common.base.Objects;</span></del>

    public class MyClass {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    package com.devonfw.ide.sonarqube.dataaccess.api;

    <u style='color: #979696'><span style='color: #648FFF'>import java.util.Objects;</span></u>

    public class MyClass {}
</pre>