<p>
    This rule checks that datatypes are mapped to the JPA using the correct pattern.
    The use of the org.hibernate.Type and javax.persistence.Convert annotations is forbidden.
    Instead use the javax.persistence.Convert annotation on a custom converter implementing the javax.persistence.AttributeConverter.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.logic.impl;

    <del style='color: #979696'><span style='color: #FFB000'>import org.hibernate.Type;</span></del>
    <del style='color: #979696'><span style='color: #FFB000'>import javax.persistence.Convert;</span></del>

    public class MyClass {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    package com.devonfw.ide.sonarqube.logic.impl;

    <u style='color: #979696'><span style='color: #648FFF'>import javax.persistence.Converter;</span></u>
    <u style='color: #979696'><span style='color: #648FFF'>import javax.persistence.AttributeConverter;</span></u>

    <u style='color: #979696'><span style='color: #648FFF'>@Converter</span></u>
    public class CustomConverter <u style='color: #979696'><span style='color: #648FFF'>implements AttributeConverter</span></u> {}
</pre>