<p>
    This rule checks that datatypes are mapped to the JPA using the correct pattern.
    The use of the org.hibernate.Type and javax.persistence.Convert annotations is forbidden.
    Instead use the javax.persistence.Convert annotation on a custom converter implementing the javax.persistence.AttributeConverter.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.logic.impl;

    <span style='color: #FFB000'><del>import org.hibernate.Type;</del></span>
    <span style='color: #FFB000'><del>import javax.persistence.Convert;</del></span>

    public class MyClass {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    package com.devonfw.ide.sonarqube.logic.impl;

    <span style='color: #648FFF'><u>import javax.persistence.Converter;</u></span>
    <span style='color: #648FFF'><u>import javax.persistence.AttributeConverter;</u></span>

    <span style='color: #648FFF'><u>@Converter</u></span>
    public class CustomConverter <span style='color: #648FFF'><u>implements AttributeConverter</u></span> {}
</pre>