<p>
    This rule checks that proper date types are used.
    The use of the java.util.Date, java.util.Calendar, java.sql.Date and java.sql.Timestamp is discouraged.
    Instead use the java.time.LocalDate[Time], java.time.ZonedDateTime, java.time.OffsetDateTime or java.time.Instant.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.dataaccess.impl;

    <span style='color: #FFB000'><del>import java.util.Date;</del></span>
    <span style='color: #FFB000'><del>import java.util.Calendar;</del></span>
    <span style='color: #FFB000'><del>import java.sql.Date;</del></span>
    <span style='color: #FFB000'><del>java.sql.Timestamp;</del></span>

    public class MyClass {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    package com.devonfw.ide.sonarqube.dataaccess.impl;

    <span style='color: #648FFF'><u>import java.time.LocalDate;</u></span>
    <span style='color: #648FFF'><u>import java.time.LocalDateTime;</u></span>
    <span style='color: #648FFF'><u>import java.time.ZonedDateTime;</u></span>
    <span style='color: #648FFF'><u>import java.time.OffsetDateTime;</u></span>
    <span style='color: #648FFF'><u>import java.time.Instant;</u></span>

    public class MyClass {}
</pre>