<p>
    This rule checks that Immutable is used from proper package in Entity class.
    The use of the javax.annotation.concurrent.Immutable annotation in Entity class is forbidden.
    Instead use the org.hibernate.annotations.Immutable.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.dataaccess.impl;

    <u style='color: #979696'><span style='color: #FFB000'>import javax.persistence.Entity;</span></u>
    <del style='color: #979696'><span style='color: #FFB000'>import javax.annotation.concurrent.Immutable;</span></del>

    @Entity
    public class FooEntity {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    package com.devonfw.ide.sonarqube.dataaccess.impl;

    <u style='color: #979696'><span style='color: #648FFF'>import javax.persistence.Entity;</span></u>
    <u style='color: #979696'><span style='color: #648FFF'>import org.hibernate.annotations.Immutable;</span></u>

    @Entity
    public class FooEntity {}
</pre>