<p>
    This rule checks that batch layer does not depend on dataaccess layer.
    Allowed dependencies from the batch layer:
    <ul type='disc'>
        <li>Batch layer of a different component</li>
        <li>Logic layer of the same component</li>
    </ul>
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.<span style='color: #785EF0'>batch</span>.api;

    import com.devonfw.ide.sonarqube.<del style='color: #979696'><span style='color: #FFB000'>dataaccess</span></del>.api.Bar;
    
    public class MyClass {}
</pre>