<p>
    This rule checks that dataaccess layer does not depend on logic layer.
    From the dataaccess layer, only dependencies to the common layer are allowed.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.<span style='color: #785EF0'>dataaccess</span>.api;

    import com.devonfw.ide.sonarqube.<span style='color: #FFB000'><del>logic</del></span>.api.Bar;
    
    public class MyClass {}
</pre>