<p>This rule checks that common layer does not depend on another layer.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.<span style='color: #785EF0'>common</span>.api;

    import com.devonfw.ide.sonarqube.<span style='color: #FFB000'><del>client</del></span>.api.Foo;
    import com.devonfw.ide.sonarqube.<span style='color: #FFB000'><del>service</del></span>.api.Bar;
    import com.devonfw.ide.sonarqube.<span style='color: #FFB000'><del>logic</del></span>.api.Some;
    
    public class MyClass {}
</pre>