<p>This rule checks that none of the other layers depend on client layer.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.<span style='color: #785EF0'>service</span>.api;

    import com.devonfw.ide.sonarqube.<del style='color: #979696'><span style='color: #FFB000'>client</span></del>.api.Bar;
    
    public class MyClass {}
</pre>