<p>This rule checks that types inheriting from Cto (Composite-Transfer-Object) also have the suffix Cto.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public class <span style='color: #FFB000'><del>MyClass</del></span> extends AbstractCto {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    public class <span style='color: #648FFF'><u>MyCto</u></span> extends AbstractCto {}
</pre>