<p>This rule checks types extending from Dao (Data-Access-Object) also start or end with Dao.
Further if they are non-abstract they should carry the suffix and DaoImpl and implement an interface with their name excluding the Impl suffix.
</p>
<h2>Noncompliant Code Examples</h2>
<pre>
    public abstract class <span style='color: #FFB000'><del>ExampleClass</del></span> extends AbstractDao {}
</pre>
<pre>
    public class <span style='color: #FFB000'><del>ExampleDaoImpl</del></span> extends AbstractDao {}
</pre>
<h2>Compliant Solutions</h2>
<pre>
    public abstract class <span style='color: #648FFF'><u>AbstractExampleDaoImpl</u></span> extends AbstractDao {}
</pre>
<pre>
    public class <span style='color: #648FFF'><u>CustomerDaoImpl</u></span> extends AbstractDao implements CustomerDao {}
</pre>