<p>
    This rule checks that non-abstract classes extending AbstractDao end with DaoImpl.
    In addition, classes that inherit from a class with the suffix Dao must also have the suffix Dao.
</p>
<h2>Noncompliant Code Examples</h2>
<pre>
    public abstract class <del style='color: #979696'><span style='color: #FFB000'>ClassDaoImpl</span></del> extends AbstractDao {}
</pre>
<pre>
    public class <del style='color: #979696'><span style='color: #FFB000'>TestClass</span></del> extends TestDao {}
</pre>
<h2>Compliant Solutions</h2>
<pre>
    public class <u style='color: #979696'><span style='color: #648FFF'>ClassDaoImpl</span></u> extends AbstractDao {}
</pre>
<pre>
    public class <u style='color: #979696'><span style='color: #648FFF'>ClassDao</span></u> extends TestDao {}
</pre>