<p>This rule checks that types inheriting from Entity end with the suffix Entity.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public class <span style='color: #FFB000'><del>ExampleClass</del></span> extends AbstractEntity {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    public class <span style='color: #648FFF'><u>ExampleEntity</u></span> extends AbstractEntity {}
</pre>