<p>This rule checks that types inheriting from Eto (Entity-Transfer-Object) end with the suffix Eto.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public class <span style='color: #FFB000'><del>ExampleClass</del></span> extends AbstractEto {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    public class <span style='color: #648FFF'><u>ExampleEto</u></span> extends AbstractEto {}
</pre>