<p>This rule checks that classes inheriting from classes with the suffix SearchCriteriaTo end with the same suffix.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public class <del style='color: #979696'><span style='color: #FFB000'>TestClass</span></del> extends AbstractSearchCriteriaTo {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    public class <u style='color: #979696'><span style='color: #648FFF'>TestSearchCriteriaTo</span></u> extends AbstractSearchCriteriaTo {}
</pre>