<p>This rule checks that types inheriting from SearchCriteriaTo end with the same suffix.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public class <span style='color: #FFB000'><del>ExampleClass</del></span> extends AbstractSearchCriteriaTo {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    public class <span style='color: #648FFF'><u>ExampleSearchCriteriaTo</u></span> extends AbstractSearchCriteriaTo {}
</pre>