<p>This rule checks that classes extending classes with the suffix To end with the same suffix.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public class <del style='color: #979696'><span style='color: #FFB000'>TestClass</span></del> extends AbstractTo {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    public class <u style='color: #979696'><span style='color: #648FFF'>UserTo</span></u> extends AbstractTo {}
</pre>