<p>This rule checks that types extending To (Transfer-Object) end with the same suffix.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public class <span style='color: #FFB000'><del>TestClass</del></span> extends AbstractTo {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    public class <span style='color: #648FFF'><u>UserTo</u></span> extends AbstractTo {}
</pre>