<p>This rule checks that ...</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public interface <span style='color: #FFB000'><del>Example</del></span> extends Foo {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    public interface <span style='color: #648FFF'><u>Example</u></span> extends Foo {}
</pre>