<p>This rule checks that types inheriting from Repository end with the same suffix.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public interface <span style='color: #FFB000'><del>TestInterface</del></span> extends DefaultRepository {}
</pre>
<h2>Compliant Solution</h2>
<pre>
    public interface <span style='color: #648FFF'><u>UserRepository</u></span> extends DefaultRepository {}
</pre>