<p>
    This rule checks that the naming of packages adheres to the devonfw package conventions. 
    For more information on the devonfw naming conventions for packages, visit 
    <a href="https://github.com/devonfw/devon4j/blob/develop/documentation/coding-conventions.asciidoc#packages">our coding conventions</a>
</p>
<h2>Noncompliant Code Examples</h2>
<pre>
    <del style='color: #979696'><span style='color: #FFB000'>package com.devonfw.ide.sonarqube.persistence.api;</span></del>
</pre>
<pre>
    <del style='color: #979696'><span style='color: #FFB000'>package com.devonfw.ide.sonarqube;</span></del>
</pre>
<pre>
    <del style='color: #979696'><span style='color: #FFB000'>package com.devonfw.ide.sonarqube.common;</span></del>
</pre>
<h2>Compliant Solution</h2>
<pre>
    <u style='color: #979696'><span style='color: #648FFF'>package com.devonfw.ide.sonarqube.bookingmanagement.dataaccess.api;</span></u>
</pre>