<p>
    This rule checks that the naming of packages adheres to the devonfw package conventions. 
    For more information on the devonfw naming conventions for packages, visit 
    <a href="https://github.com/devonfw/devon4j/blob/develop/documentation/coding-conventions.asciidoc#packages">our coding conventions</a>
</p>
<h2>Noncompliant Code Examples</h2>
<pre>
    <span style='color: #FFB000'><del>package com.devonfw.ide.sonarqube.persistence.api;</del></span>
</pre>
<pre>
    <span style='color: #FFB000'><del>package com.devonfw.ide.sonarqube;</del></span>
</pre>
<pre>
    <span style='color: #FFB000'><del>package com.devonfw.ide.sonarqube.common;</del></span>
</pre>
<h2>Compliant Solution</h2>
<pre>
    <span style='color: #648FFF'><u>package com.devonfw.ide.sonarqube.bookingmanagement.dataaccess.api;</u></span>
</pre>