<p>
    This rule checks that api scope does not depend on base scope.
    Api scope is only allowed to depend on the api scope of another component.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.common.<span style='color: #785EF0'>api</span>;

    import com.devonfw.ide.sonarqube.common.<del style='color: #979696'><span style='color: #FFB000'>base</span></del>.Bar;
    
    public class MyClass {}
</pre>