<p>
    This rule checks that base scope does not depend on the impl scope of the same component.
    Allowed dependencies from base scope:
    <ul type='disc'>
        <li>Api scope of the same component</li>
        <li>Api scope of a different component</li>
        <li>Base scope of a different component</li>
    </ul>
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.<span style='color: #785EF0'>component1.logic.base</span>;

    import com.devonfw.ide.sonarqube.<del style='color: #979696'><span style='color: #FFB000'>component1.logic.impl</span></del>.Bar;

    public class MyClass {}
</pre>