<p>
    This rule checks that impl scope does not depend on impl scope of another component part.
    Impl scope is only allowed to depend on the base scope of the same component.
</p>
<h2>Noncompliant Code Example</h2>
<pre>
    package com.devonfw.ide.sonarqube.<span style='color: #785EF0'>component1.service.impl</span>;

    import com.devonfw.ide.sonarqube.<del style='color: #979696'><span style='color: #FFB000'>component2.service.impl</span></del>.Bar;

    public class MyClass {}
</pre>